python/ray/data/read_api.py

```diff
@@ -3009,6 +3011,89 @@ def from_torch(
     )
 
 
+@PublicAPI
+def read_iceberg(
+    *,
+    table_identifier: str,
+    row_filter: Union[str, "BooleanExpression"] = None,
+    parallelism: int = -1,
+    selected_fields: Tuple[str, ...] = ("*",),
+    snapshot_id: Optional[int] = None,
+    scan_kwargs: Optional[Dict[str, str]] = None,
+    catalog_kwargs: Optional[Dict[str, str]] = None,
+    ray_remote_args: Optional[Dict[str, Any]] = None,
+    override_num_blocks: Optional[int] = None,
+) -> Dataset:
+    """Create a :class:`~ray.data.Dataset` from an Iceberg table. The table to read
+    from is specified using a fully qualified ``table_identifier``. Using PyIceberg,
+    any intended row filters, snapshot IDs, etc. are applied, and the files that
+    satisfy the query are distributed across Ray read tasks. The number of output
+    blocks is determined by ``override_num_blocks`` which can be requested from
+    this interface or automatically chosen if unspecified.
```

Comment:
nit: for better docs/docstring formatting (also on IDE views), the first sentence of the docstring should be separated into its own line.
```suggestion
    """Create a :class:`~ray.data.Dataset` from an Iceberg table. 
    
    The table to read from is specified using a fully qualified ``table_identifier``. Using PyIceberg,
    any intended row filters, snapshot IDs, etc. are applied, and the files that
    satisfy the query are distributed across Ray read tasks. The number of output
    blocks is determined by ``override_num_blocks`` which can be requested from
    this interface or automatically chosen if unspecified.
```